src/Middleware/RecordTransaction.php

```diff
@@ -44,7 +45,7 @@ public function handle(Request $request, Closure $next)
         // Execute the application logic
         $response = $next($request);
 
-        if (config('elastic-apm-laravel.transactions.useRouteUri')) {
+        if (Config::get('elastic-apm-laravel.transactions.useRouteUri')) {
```

Comment:
I'm having some trouble to make `Config::get` return `true` in one specific spec. I'm using `_before` to always return `false`:

```php
protected function _before()
{
    Config::shouldReceive('get')
        ->with('elastic-apm-laravel.transactions.useRouteUri')
        ->andReturn(false);
    ...
}
```

But in one specific spec I want to return `true`:

```php
public function testUriTransactionName()
{
    Config::shouldReceive('get')
        ->with('elastic-apm-laravel.transactions.useRouteUri')
        ->andReturn(true);
   ...
    $this->middleware->handle($this->request, function () {
        return $this->response;
    });
}
```

@cykirsch any idea how to achieve that with Facades?

Reply:
I haven't actually tried to do that pattern, and was surprised to see it works for you in some of the other tests I've looked at in the package. So no, I don't know how to achieve it with facades.

I'd side with @countless-integers https://github.com/arkaitzgarro/elastic-apm-laravel/pull/29/files#r383183831. Prevents any need for workarounds in both of these cases, and it's very explicit. If the duplication is bothersome it can go in a helper method so it's at least a shorter call to make.